diff-match-patch/diff-match-patch.d.ts

```diff
@@ -0,0 +1,43 @@
+// Type definitions for diff-match-patch v1.0.0
+// Project: https://www.npmjs.com/package/diff-match-patch
+// Definitions by: Austen Talbot <https://github.com/austentalbot/>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+declare module "diff-match-patch" {
+    type Diff = [number, string];
+
+    export class DiffMatchPatch {
+    	Diff_Timeout: number;
+    	Diff_EditCost: number;
+    	Match_Threshold: number;
+    	Match_Distance: number;
+    	Patch_DeleteThreshold: number;
+    	Patch_Margin: number;
+    	Match_MaxBits: number;
+
+    	diff_main(text1: string, text2: string, opt_checklines?: boolean, opt_deadline?: number): Diff[];
+    	diff_commonPrefix(text1: string, text2: string): number;
+    	diff_commonSuffix(text1: string, text2: string): number;
+    	diff_cleanupSemantic(diffs: Diff[]): void;
+    	diff_cleanupSemanticLossless(diffs: Diff[]): void;
+    	diff_cleanupEfficiency(diffs: Diff[]): void;
+    	diff_cleanupMerge(diffs: Diff[]): void;
+    	diff_xIndex(diffs: Diff[], loc: number): number;
+    	diff_prettyHtml(diffs: Diff[]): string;
+    	diff_text1(diffs: Diff[]): string;
+    	diff_text2(diffs: Diff[]): string;
+    	diff_levenshtein(diffs: Diff[]): number;
+    	diff_toDelta(diffs: Diff[]): string;
+    	diff_fromDelta(text1: string, delta: string): Diff[];
+
+        new (): DiffMatchPatch;
+    }
+
+    export var DIFF_DELETE: number;
+    export var DIFF_INSERT: number;
+    export var DIFF_EQUAL: number;
+
+    export var diff_match_patch: {
+    	new (): DiffMatchPatch;
```

Comment:
So this library exposes both `DiffMatchPatch` and `diff_match_path`? Seems instead like the class can just be named `diff_match_path`?